beacon-chain/sync/validate_beacon_blocks.go

```diff
@@ -343,6 +351,24 @@ func (s *Service) validateBellatrixBeaconBlock(ctx context.Context, parentState
 	return nil
 }
 
+// Verifies the signature of the pending block with respect to the current head state.
+func (s *Service) verifyPendingBlockSignature(ctx context.Context, blk interfaces.ReadOnlySignedBeaconBlock, blkRoot [32]byte) (pubsub.ValidationResult, error) {
+	roState, err := s.cfg.chain.HeadStateReadOnly(ctx)
+	if err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	// Ignore block in the event of non-existent proposer.
+	_, err = roState.ValidatorAtIndex(blk.Block().ProposerIndex())
+	if err != nil {
+		return pubsub.ValidationIgnore, err
+	}
+	if err := blocks.VerifyBlockSignatureUsingCurrentFork(roState, blk); err != nil {
```

Comment:
Do you also want to process the state to the current block slot?

Reply:
Because we are verifying signatures, I don't think we need to do this. You just need the correct domain( fork), pubkey and signature to verify the block correctly. If you feel strongly, I can add this in

Reply:
The issue here is if we recieve a block across fork, we will not be able to verify it or insert it to the pending queue . That's a step back from the current status quo

Reply:
My previous comment needed to be corrected. `VerifyBlockSignatureUsingCurrentFork` does not verify the using current state but using slot to derive the fork schedule